app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -1061,6 +1063,19 @@ class MainViewModel @Inject constructor(
         return messageController.getMessageCountInThreadForFolder(threadUid, folderId, mailboxContentRealm()) == 1L
     }
 
+    fun shareThreadUrl(messageUid: String, activityContext: Context) {
+        val message = messageController.getMessage(messageUid) ?: return
+        val mailboxUuid = currentMailbox.value?.uuid ?: return
+
+        viewModelScope.launch(ioCoroutineContext) {
+            val response = ApiRepository.getShareLink(mailboxUuid, message.folderId, message.shortUid)
+
+            if (response.isSuccess() && response.data != null) {
+                response.data!!.url.let { activityContext.shareString(it) }
+            }
+        }
+    }
+
```

Comment:
You have a code smell here. If message is `null`, you initialize a `val` for nothing.

```suggestion
fun shareThreadUrl(messageUid: String, activityContext: Context) {
        messageController.getMessage(messageUid)?.let { message ->
            val mailboxUuid = currentMailbox.value?.uuid ?: return

            viewModelScope.launch(ioCoroutineContext) {
                val response = ApiRepository.getShareLink(mailboxUuid, message.folderId, message.shortUid)

                if (response.isSuccess() && response.data != null) {
                    activityContext.shareString(response.data!!.url)
                }
            }
        }
    }
```